calico_containers/pycalico/ipam.py

```diff
@@ -782,6 +820,56 @@ def get_assignment_attributes(self, address):
             _, attributes = block.get_attributes_for_ip(address)
             return attributes
 
+    def get_cidr_assignment_attributes(self, cidr):
+        """
+        Return the attributes of a given CIDR. Depending on the size of the
+        CIDR in question, this could have a very large result.
+
+        :param cidr: CIDR to query.
+        :type cidr: IPNetwork
+        :return: The attributes of all the addresses in a given CIDR.
+        :rtype: list
+        """
+        assert isinstance(cidr, IPNetwork)
+
+        # List of IPNetworks corresponding to blocks in ETCD
+        block_cidrs = self._get_all_blocks(cidr)
+
+        # List of AllocationBlocks
+        blocks = []
+        for block_cidr in block_cidrs:
+            block = self._read_block(block_cidr)
+            blocks.append(block)
+
+        # List of IPPools
+        pools = self.get_ip_pools(cidr.version)
+
+        result = []
+        for block in blocks:
+            for pool_candidate in pools:
+                if block in pool_candidate:
+                    pool = str(pool_candidate)
+                    break
+
+            for num, index in enumerate(block.allocations):
+                ip_int = block.first + num
+                if cidr.first <= ip_int <= cidr.last:
+                    ip_attrs = {}
+                    ip_attrs["POOL"] = pool
+                    if index is None:
+                        ip_attrs["ALLOCATED"] = False
+                        ip_attrs["ATTRIBUTES"] = {}
+                        ip_attrs["HANDLE"] = None
+                    else:
+                        attributes = block.attributes[index]
+                        ip_attrs["ATTRIBUTES"] = attributes["secondary"]
+                        ip_attrs["HANDLE"] = attributes["handle_id"]
+                        ip_attrs["IP"] = str(IPAddress(ip_int))
+                    result.append(ip_attrs)
```

Comment:
The size of this data could certainly get quite large. I was debating whether it would be better to leave the objects as is (i.e. pool = IPPool,  ip = IPAddress), or converting them to strings was the better option. Converting them back for the consumer could get expensive, but if we leave as is, the result is large.

Last night I was thinking strings would be cleaner, because my original intent was User read reports, however this morning I was thinking that the data transfer of these large results is already done. I.e. the damage is already done, memory is allocated and the datastore has already been hit, so why not return the whole objects and let the consumer decide what to do with it.

The only other use case that comes to mind, is a return bulk update operation. The consumer could use this get the objects they need, update them, then send them back.